worker/fetchEmbedMeta.js

```diff
@@ -0,0 +1,92 @@
+import { parseEmbedUrl, findUniqueUrls } from '@/lib/url'
+import { getEmbedMetaFetcher } from '@/lib/embedFetcher'
+
+export async function fetchEmbedMeta ({ data: { id: itemId }, models }) {
+  console.log('[fetchEmbedMeta] fetching embed meta for item', itemId)
+  const item = await models.item.findUnique({ where: { id: itemId } })
+
+  const text = item.text || item.url
+  if (!text) return
+
+  const urls = findUniqueUrls(text) // extract every url from the text
+  if (!urls.length) return
+
+  const embeds = urls
+    .map(url => parseEmbedUrl(url)) // parse every unique url as a potential embed (initial url based deduplication)
+    .filter(embed => !!embed) // if it is not an embed, we can skip it
+    .reduce((acc, embed) => { // deduplicate embeds by id and provider (they might have different urls but be resolved to the same embed)
```

Comment:
This is to avoid to fetch the same thing more than once.

> Why would someone want to embed the same thing more than once? 

Idk to abuse the system, or meme, either way, better to stop it here rather than firing multiple fetches